### PR TITLE
wallet rpc: return output keys with incoming transfers

### PR DESCRIPTION
**Pull request by benevanoff**
Mentioned in IRC

### CODE REVIEW

**Review by ndorf — APPROVED** [review]

File: src/wallet/wallet_rpc_server.cpp [hunk 1]
```diff
@@ -1888,6 +1888,13 @@ namespace tools
         rpc_transfers.tx_hash      = epee::string_tools::pod_to_hex(td.m_txid);
         rpc_transfers.subaddr_index = {td.m_subaddr_index.major, td.m_subaddr_index.minor};
         rpc_transfers.key_image    = td.m_key_image_known ? epee::string_tools::pod_to_hex(td.m_key_image) : "";
+        std::vector<std::string> keys;
+        for (auto o : td.m_tx.vout)
```
Comment (mj-xmr) [inline src/wallet/wallet_rpc_server.cpp:1892]: Can it be const auto & o ?

File: src/wallet/wallet_rpc_server.cpp [hunk 2]
```diff
@@ -1888,6 +1888,13 @@ namespace tools
         rpc_transfers.tx_hash      = epee::string_tools::pod_to_hex(td.m_txid);
         rpc_transfers.subaddr_index = {td.m_subaddr_index.major, td.m_subaddr_index.minor};
         rpc_transfers.key_image    = td.m_key_image_known ? epee::string_tools::pod_to_hex(td.m_key_image) : "";
+        std::vector<std::string> keys;
+        for (auto o : td.m_tx.vout) 
+        {
+          std::string key = epee::string_tools::pod_to_hex(boost::get<cryptonote::txout_to_key>(o.target).key);
```
Comment (mj-xmr) [inline src/wallet/wallet_rpc_server.cpp:1894]: This can be const for sure.

Comment (mrtestyboy781) [inline src/wallet/wallet_rpc_server.cpp:1894]: In the similar wallet-cli command it actually only lists one output per key image, as I read, this is listing all the vouts. I believe we can filter this to only be the single vout matching the key image

Comment (jorgeelmundoso) [reply src/wallet/wallet_rpc_server.cpp:1894]: I guess what should be taken care of is to make sure if the get_transfers call is filtered by account.major/minor only the correlating vout/vouts should be returned.

**Comment by benevanoff** [comment]
@mrtestyboy781 I've changed it to only return the owned public key associated with that key image :)

**Comment by mrtestyboy781** [comment]
Tested and lgtm, thank you @benevanoff

**Review by moneromooo-monero — APPROVED** [review]

**Pull request merged**